src/js/player.js

```diff
@@ -2928,11 +2928,50 @@ class Player extends Component {
     }
     this.loadTech_(this.options_.techOrder[0], null);
     this.techCall_('reset');
+    this.resetControlBarUI();
     if (isEvented(this)) {
       this.trigger('playerreset');
     }
   }
 
+  /**
+   * Reset Control Bar's UI by calling sub-methods that reset
+   * all of Control Bar's components
+   */
+  resetControlBarUI() {
+    this.resetProgressBar_();
+    this.resetPlaybackRate_();
+    this.resetVolumeBar_();
+  }
+
+  /**
+   * Reset tech's progress so progress bar is reset in the UI
+   */
+  resetProgressBar_() {
+    this.currentTime(0);
+    if (isNaN(this.duration())) {
+      this.duration(0);
+    }
+    this.tech_.trigger({ type: 'durationchange', target: this.tech_, manuallyTriggered: true });
```

Comment:
According to the spec, we shouldn't be triggering a `durationchange` if the duration is NaN.